defaults/main.yml

```diff
@@ -1,7 +1,32 @@
+# note: the booleans are quoted otherwise they are rendered capitalised.
 ---
+buildkite_agent_major_version: 3
 buildkite_agent_conf_dir: "/etc/buildkite-agent"
 buildkite_agent_token: ""
-buildkite_agent_name: "buildkite-agent"
+buildkite_agent_debug: "false"
+buildkite_agent_enabled: true
+# if you're building a golden image and don't want it to connect to
+# buildkite during build ptocess, set this to 'stopped'.
+buildkite_agent_state: "started"
+
+# v2 https://buildkite.com/docs/agent/v2/configuration
+buildkite_agent_name: "buildkite-agent" # v3 default: '%hostname-%n'
 buildkite_agent_meta_data: "queue=default"
 buildkite_agent_meta_data_ec2_tags: "true"
-buildkite_agent_debug: "false"
+
+# v3 https://buildkite.com/docs/agent/v3/configuration
+buildkite_agent_priority: 1
+buildkite_agent_queue: "default"
```

Comment:
Since this only gets used in `buildkite_agent_tags`, do you think we could get rid of the variable entirely and just leave  `buildkite_agent_tags` with a `queue=` default.

Reply:
I'd like to not do that, because of how I'm intending to roll out changes to our own buildkite cluster. 

I am planning to use a git sha1 from my config-management-and-packer repo as the value of the `queue` tag - this means that I know the provenance of the agents, and also means that when I want to make changes I can do so in a safe way because new agents get assigned to an isolated queue. Pipeline steps can then be transitioned across by changing their queue to that of the new agent cluster.

I also thought keeping it separate makes a bit of logical sense since it's a special case for a tag.

What do you think?

Reply:
Seems fine to keep then.

Reply:
:+1: